### ISSUE TITLE
CLI error when running config with two files with the same name in different subfolders

### ISSUE THREAD

**Issue by lavoiedn** [datawire/ambassador#494]
**Describe the bug**
I'm running into an issue while trying to run `ambassador config <k8s_manifest_path> config.json --k8s` with ambassador cli. If I have two subfolders that contain a yaml with the same name, I get the following error:

```
2018-06-06 13:43:16 ambassador 0.34.1 ERROR: EXCEPTION from parse_config: '<' not supported between instances of 'dict' and 'dict'
Traceback (most recent call last):

  File "/home/user/git/ambassador/venv/lib/python3.6/site-packages/ambassador-0.34.1-py3.6.egg/ambassador/cli.py", line 92, in parse_config
    template_dir_path=template_dir_path, schema_dir_path=schema_dir_path)

  File "/home/user/git/ambassador/venv/lib/python3.6/site-packages/ambassador-0.34.1-py3.6.egg/ambassador/config.py", line 336, in __init__
    self.process_all_objects()

  File "/home/user/git/ambassador/venv/lib/python3.6/site-packages/ambassador-0.34.1-py3.6.egg/ambassador/config.py", line 401, in process_all_objects
    for filename, ocount, obj in sorted(self.objects_to_process):

TypeError: '<' not supported between instances of 'dict' and 'dict'
```

**To Reproduce**
Prepare some configuration with a file structure something like this:
```
ambassador-config
└ default
│   └ ambassador.yaml
│   └ qotm.yaml
└ namespace
    └ qotm.yaml
```

I have attached an archive containing the files I used for testing:
[ambassador-example-config.tar.gz](https://github.com/datawire/ambassador/files/2077616/ambassador-example-config.tar.gz)

Then, run:
`ambassador config </path/to/config> ./config.json --k8s`

**Expected behavior**
I expect configuration to support multiple configuration files with the same name on different levels in the file structure.

**Versions (please complete the following information):**
 - Ambassador: 0.34.1
 - Kubernetes environment: N/A (ambassador cli by itself)
 - Version: N/A

**Additional context**
I think this is due to how the objects to sort are built, they use (filename, ocount, obj) so when trying to sort, if `filename` and `ocount` are the same, it falls back to comparing `object_to_process_a[2] == object_to_process_b[2]` (`obj`) which are both dicts and fails with the above error.

Is there any issue with changing the tuple's format from `(filename, ocount, obj)` to `(filepath, ocount, obj)`? From what I've seen this does not cause any issues but I must admit I did not try running it in a complete k8s test environment. I'll open a PR with this solution, feel free to close it if this does not work with a full deployment.

**Comment by lavoiedn** [comment]
Note that, after working on this issue for a bit it seems like there is a larger underlying issue where generating the envoy config from the cli does not work the same way as generating the config in `kubewatch.py` does.

I'm not sure what the intended usage of the CLI is in relation to the config but we've been using it to validate that there were no errors in the ambassador configuration in our kubernetes manifests using the `validate` function.

However, some issues recently arose when generating config from the cli using kubernetes resources yamls:

- Does not take `AMBASSADOR_NAMESPACE` or `AMBASSADOR_SINGLE_NAMESPACE` into account
- Does not parse the yaml resources' `metadata.name` or `metadata.namespace`

This also raises some questions on proper configuration, such as:

- Should mapping names be unique across namespaces even when running one ambassador per namespace?
- What about modules that are name dependant?

**Comment by kflynn** [comment]
Fixed by #520.
